libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/targets/native/spm/SwiftPackageExtension.kt

```diff
@@ -0,0 +1,380 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.gradle.targets.native.spm
+
+import groovy.lang.Closure
+import org.gradle.api.Project
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Nested
+import org.gradle.api.tasks.Optional
+import org.gradle.util.ConfigureUtil
+
+class SwiftPackageExtension(private val project: Project) {
+    @Input
+    lateinit var name: String
+
+    // TODO: List<String> or String
+    @Input
+    @Optional
+    var defaultLocalization: List<String>? = null
+
+    @Nested
+    @Optional
+    var platforms: List<SupportedPlatform>? = null
+
+    @Nested
+    var productsLibraries: List<Products.ProductType.Library> = listOf()
+
+    @Nested
+    var productsExecutables: List<Products.ProductType.Executable> = listOf()
+
+    @Nested
+    var dependencies: List<Dependencies.Package> = listOf()
+
+    @Nested
+    var targetsRegular: List<Targets.Target> = listOf()
+
+    @Nested
+    var targetsExecutable: List<Targets.Target> = listOf()
+
+    @Nested
+    var targetsTest: List<Targets.Target> = listOf()
+
+    // TODO: swiftLanguageVersions: [SwiftVersion]? = nil,
+    // TODO: cLanguageStandard: CLanguageStandard? = nil,
+    // TODO: cxxLanguageStandard: CXXLanguageStandard? = nil
+
+    fun platforms(configure: SupportedPlatform.() -> Unit) {
+        val supportedPlatform = SupportedPlatform().apply(configure)
+        platforms = supportedPlatform.platforms
+    }
+
+    fun products(configure: Products.() -> Unit) {
+        val product = Products().apply(configure)
+
+        productsLibraries = product.libraries
+        productsExecutables = product.executables
+    }
+
+    fun dependencies(configure: Dependencies.() -> Unit) {
+        val dependency = Dependencies().apply(configure)
+        dependencies = dependency.dependencies
+    }
+
+    fun targets(configure: Targets.() -> Unit) {
+        val target = Targets().apply(configure)
+
+        targetsRegular = target.regulars
+        targetsExecutable = target.executables
+        targetsTest = target.tests
+    }
+
+    // TODO: add fun with another params (supportedPlatform)
+    class SupportedPlatform() {
```

Comment:
Please, try to rethink this part. `Supported` is just the attribute of platform, but here `SupportedPlatform` is responsible for holding info like container and could be changed by its function calls.

Reply:
Take a look at org.jetbrains.kotlin.gradle.plugin.KotlinPlatformType and its usages (for example, org/jetbrains/kotlin/gradle/plugin/mpp/KotlinMultiplatformPlugin.kt:302)